scheduler-worker/src/test/java/gov/nasa/jpl/aerie/scheduler/worker/services/SchedulingIntegrationTests.java

```diff
@@ -486,6 +486,41 @@ export default () => Goal.CoexistenceGoal({
     }
   }
 
+  @Test
+  void testCoexistenceGoalWithAnchorsCreation() {
+    final var results = runScheduler(
+        BANANANATION,
+        List.of(
+            new ActivityDirective(
+                Duration.ZERO,
+                "BiteBanana",
+                Map.of("biteSize", SerializedValue.of(1)),
+                null,
+                true
+            )
+        ),
+        List.of(new SchedulingGoal(new GoalId(0L, 0L), """
+          export default () => Goal.CoexistenceGoal({
+            persistentAnchor: PersistentTimeAnchor.START,
+            forEach: ActivityExpression.ofType(ActivityTypes.BiteBanana),
+            activityFinder: ActivityExpression.ofType(ActivityTypes.GrowBanana),
+            activityTemplate: (interval) => ActivityTemplates.GrowBanana({quantity: 10, growingDuration: Temporal.Duration.from({minutes:1}) }),
+            startsAt: TimingConstraint.singleton(WindowProperty.END).plus(Temporal.Duration.from({ minutes : 5}))
+          })
+          """, true)),
+        PLANNING_HORIZON);
+
+    assertEquals(1, results.scheduleResults.goalResults().size());
+    final var goalResult = results.scheduleResults.goalResults().get(new GoalId(0L, 0L));
+
+    assertTrue(goalResult.satisfied());
+    assertEquals(1, goalResult.createdActivities().size());
+    assertEquals(1, goalResult.satisfyingActivities().size());
+    for (final var activity : goalResult.satisfyingActivities()) {
+      assertNotNull(activity);
```

Comment:
This test ought to assert that the scheduler created the anchor it was supposed to.